src/app/utils/tests/TestUtils.spec.ts

```diff
@@ -715,16 +715,28 @@ export async function expectPermissionToBeDenied<T>(promise: Promise<T>): Promis
     await promise.then(throwIfFulfilled, checkErrorCode);
 }
 
+/**
+ * Returns a checker to verify that a subscription method has been correctly unsubscribed
+ * to in case it has been subscribed first.
+ */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export function prepareUnsubscribeCheck(service: any, subscribeMethod: string): () => void {
 
+    let subscribed: boolean = false;
     let unsubscribed: boolean = false;
-    spyOn(service, subscribeMethod).and.returnValue(new Subscription(() => {
-        unsubscribed = true;
-    }));
+    spyOn(service, subscribeMethod).and.callFake(() => {
+        subscribed = true;
+        return new Subscription(() => {
+            unsubscribed = true;
+        });
+    });
     return () => {
-        expect(unsubscribed)
-            .withContext('Service should have unsubscribed to ' + subscribeMethod + ' method bub did not')
-            .toBeTrue();
+        if (subscribed) {
```

Comment:
assert that it is subscribed instead of if (true) ?

Reply:
non, on veut checker la formule logique "tu as souscrit -> tu as désouscrit", qui est valide si:
  - tu as souscrit et désouscrit
  - tu n'as pas souscrit
Ce qui correspond bien au code !

Reply:
Oui mais le else n'as pas d'expect, donc pas de cassage de test

Reply:
oui exactement, si tu n'as pas souscrit, tu n'as pas à désouscrire, donc pas de cassage de test ! 
a -> b implique que:

  - si a et b, a->b est vrai logiquement, donc pas cassage de test
  - si a et pas b, a->b est faux logiquement, alors cassage de test
  - si pas a, a->b est vrai logiquement, alors PAS cassage de test

Reply:
c'est vrai, mais pourquoi 100% des tests qui utilisent cette méthode commencent par "should unsubscribe" si ils sont pas expectés à subscribe aussi :p Monsieur Gorge Boule ?

Reply:
je dirais que c'est deux choses différentes : ici on veut pas leaker (c'est donc un "unsubscribe check"), si on veut checker qu'on souscrit, bah c'est autre chose.
Les tests qui disent "should unsubscribe" sont des tests qui veulent dire "should not leak subscriptions". Si y'a pas de subscription, y'a rien à leak.

Tester qu'on subscribe me semble useless et pas désirable. Car si on souscrit pas, y'aura d'office de la couille dans le paté en terme de fonctionnalité !

Reply:
Non, dans la fonction tu prends déjà en compte si c'est abonné ou pas, on teste bien le désabonnement, mot pour mot, si ça s'est pas abonné, ça c'est pas désabonné DONC le test doit échouer, pas dire "ah bizarre y a pas d'expect"

Reply:
Bah et pour les cas, comme maintenant dans cette branche, ou on veut tester que "si on s'est souscrit, on se désouscrit" ? Faut deux subscription check? "prepareUnsubscribeCheckOnlyIfAlreadySubscribed" et "prepareUnconditionalUnsubscriptionCheck" ?

Reply:
hmmm ou ptet que le cas "conditional" on devrait juste s'assurer que la condition soit toujours vraie dans les tests, i.e., qu'on teste pas du non-souscrit avec le unsubscribe check? Ptet faut que je revoie du test à part-creation pour raffiner cela alors

Reply:
voilà, refactorisé pour obligé que si on utilise ce check, on doit avoir souscrit !